browser/search_engines/search_engine_provider_service_browsertest.cc

```diff
@@ -260,6 +267,64 @@ IN_PROC_BROWSER_TEST_F(SearchEngineProviderServiceTest,
 #endif
 }
 
+class SearchSuggestionsEnabledTest : public InProcessBrowserTest,
+                                     public ::testing::WithParamInterface<
+                                         std::tuple<std::string, bool, bool>> {
+ public:
+  SearchSuggestionsEnabledTest() : default_locale(GetLocale()) {}
+  ~SearchSuggestionsEnabledTest() override = default;
+
+  void SetUpCommandLine(base::CommandLine* command_line) final {
+    if (IsNewUser()) {
+      command_line->AppendSwitch(switches::kForceFirstRun);
+    }
+  }
+
+  std::string GetLocale() const { return std::get<0>(GetParam()); }
+  bool IsNewUser() const { return std::get<1>(GetParam()); }
+  bool IsSearchSuggestionsEnabled() const { return std::get<2>(GetParam()); }
+
+  const brave_l10n::test::ScopedDefaultLocale default_locale;
+};
+
+IN_PROC_BROWSER_TEST_P(SearchSuggestionsEnabledTest,
+                       DefaultSearchSuggestEnabledTest) {
+  auto* prefs = browser()->profile()->GetPrefs();
+  auto* service =
+      TemplateURLServiceFactory::GetForProfile(browser()->profile());
+  auto brave_search_data = TemplateURLDataFromPrepopulatedEngine(
+      TemplateURLPrepopulateData::brave_search);
+  auto brave_template_url = std::make_unique<TemplateURL>(*brave_search_data);
+
+  auto bing_search_data = TemplateURLDataFromPrepopulatedEngine(
+      TemplateURLPrepopulateData::brave_bing);
+  auto bing_template_url = std::make_unique<TemplateURL>(*bing_search_data);
```

Comment:
Do these have to be `std::unique_ptr<>`s?
E.g.
```cpp
TemplateURL bing_template_url(*bing_search_data);
// ...
service->SetUserSelectedDefaultSearchProvider(&bing_template_url);
```
should work just fine, too, I guess.